Assets/Shopify/UIToolkit/Themes/Generic/GenericSingleProductTheme.cs

```diff
@@ -81,6 +85,29 @@ public class GenericSingleProductTheme : MonoBehaviour, ISingleProductTheme {
             UpdateDetailsUsingVariant(wrappedVariant.ProductVariant);
         }
 
+        public void SelectImage(Button button) {
+            ActiveImage.GetComponent<UnityEngine.UI.Image>().sprite = button.image.sprite;
+        }
+
+        private void SetupPreviewImages(Product product) {
+            var images = (List<Shopify.Unity.Image>)product.images();
+            if (images.Count == 0) {
+                // TODO: Hide image area?
+                return;
+            }
+
+            ActiveImage.GetComponent<RemoteImageLoader>().LoadImage(images[0].src(), () => {
+                ActiveImage.GetComponent<UnityEngine.UI.Image>().enabled = true;
+            }, null);
+
+            foreach (var image in images) {
```

Comment:
we should only use a subset of images incase someone has like 50, lets just take the `.First(3)`?

Reply:
Alternatively we could add in a paging carousel but since this is just a generic view I'm good for limiting it to 3 for now.